controllers/cluster_controller.go

```diff
@@ -111,6 +111,16 @@ func (r *ClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request) (_
 		return ctrl.Result{}, err
 	}
 
+	if cluster.IsManaged() && cluster.Spec.BundlesRef == nil {
```

Comment:
I think we should do this later in the flow, after we check for `if cluster.IsSelfManaged()`
And that way we can avoid checking `cluster.IsManaged()`